xapian-core/tests/api_sorting.cc

```diff
@@ -321,79 +321,46 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
 }
 
 DEFINE_TESTCASE(replace_weights, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
-
     Xapian::Database mydb(get_database("apitest_onedoc"));
-
     Xapian::Enquire enquire(mydb);
-
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
-
-    // retrieve the top ten results (we only expect one)
     Xapian::MSet mymset = enquire.get_mset(0, 10);
-
-    // assigning new weight of -1 to the document.
-    vector<double> weights {-1};
+    double new_weight = (mymset.begin()).get_weight() - 0.01;
```

Comment:
I meant actually to find out what that value was, rather than calculating it. It's probably fine here, but it causes you problems in a later test, where you just set all the weights to that first (highest) original weight…